lib/collection/src/lib.rs

```diff
@@ -407,15 +442,34 @@ impl Collection {
     /// - Saves new params on disk
     /// - Stops existing optimization loop
     /// - Runs new optimizers with new params
-    pub async fn update_optimizer_params(
+    pub async fn update_optimizer_params_from_diff(
         &self,
         optimizer_config_diff: OptimizersConfigDiff,
     ) -> CollectionResult<()> {
         for shard in self.all_shards() {
-            shard
-                .get()
-                .update_optimizer_params(optimizer_config_diff.clone())
-                .await?;
+            if let Shard::Local(shard) = shard {
+                shard
+                    .update_optimizer_with_diff(optimizer_config_diff.clone())
+                    .await?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Updates shard optimization params:
+    /// - Saves new params on disk
+    /// - Stops existing optimization loop
+    /// - Runs new optimizers with new params
+    pub async fn update_optimizer_params(
+        &self,
+        optimizer_config: OptimizersConfig,
+    ) -> CollectionResult<()> {
+        for shard in self.all_shards() {
+            if let Shard::Local(shard) = shard {
+                shard
+                    .update_optimizer_with_config(optimizer_config.clone())
+                    .await?;
+            }
```

Comment:
Why parameter update is used now for local shard only?

Reply:
Consensus will ensure that this function is called on all participating peers, so all actual shards will be updated. Optimizer config as part of the collection config is the global state, so its updates will be distributed across the network.

We are using the `RemoteShard` abstraction for now only for direct requests/updates, the global state instead is planned to be updated with consensus.